zerver/views/message_edit.py

```diff
@@ -148,7 +148,7 @@ def update_message_backend(
 
 
 def validate_can_delete_message(user_profile: UserProfile, message: Message) -> None:
-    if user_profile.can_delete_any_message():
+    if user_profile.can_delete_any_message() and not user_profile.is_guest:
```

Comment:
I think we want `user_profile.can_delete_any_message` to be false for guest users, since that seems like an abstraction that might get reused.

Reply:
This setting already has `allow_everyone_group` as `false`, but the user-defined groups may contain guest users. Don't think we would want to just not allow the groups which contains at least one guest user when user changes the setting.

> I think we want user_profile.can_delete_any_message to be false for guest users, since that seems like an abstraction that might get reused.

As I suggested above. Can we have a function which checks this using `allow_everyone` for a setting and then call `is_user_in_group`. That way we can just have a single function used for all group-type settings.

Reply:
Yeah, that shared code path function approach sounds good; if `allow_everyone` is false for a group-valued setting, we should just always reject for guest users.

(Probably we'll end up renaming that to `allow_guests` for clarity later, maybe after we get a bit more of the system built out).